specification/saas/resource-manager/Microsoft.SaaS/preview/2018-03-01-beta/saas.json

```diff
@@ -356,6 +356,445 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SaaS/resources/{resourceName}": {
+      "get": {
+        "tags": [
+          "SubscriptionLevelSaaSResources"
+        ],
+        "operationId": "SaasSubscriptionLevel_Get",
+        "description": "Gets information about the specified Subscription Level SaaS.",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The request returns the resulting SaaS resource.",
+            "schema": {
+              "$ref": "#/definitions/SaasResource"
+            }
+          },
+          "default": {
+            "description": "A response indicating an error.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get saas application": {
+            "$ref": "./examples/saasSubscriptionLevel/SaasGet.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "SubscriptionLevelSaaSResources"
+        ],
+        "operationId": "SaasSubscriptionLevel_CreateOrUpdate",
+        "description": "Creates or updates a SaaS resource.",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SaasResourceCreation"
+            },
+            "description": "Parameters supplied to the create or update subscription level saas operation."
+          }
+        ],
+        "responses": {
```

Comment:
>responses [](start = 9, length = 9)

On a long-running PUT using the 202 pattern, you should also include a fake 200 response to represent what the final async response would look like. (This is a workaround for limitations of open api v2) Should be fairly easy to find example of this in the swagger repo.

Reply:
Comment applies to all occurrences of the problem with 202 responses in this swagger.

---
In reply to: [551677485](https://github.com/Azure/azure-rest-api-specs/pull/11903#discussion_r551677485) [](ancestors = 551677485)

Reply:
wilco